i2c_manager/i2c_manager.c

```diff
@@ -0,0 +1,368 @@
+/*
+
+SPDX-License-Identifier: MIT
+
+MIT License
+
+Copyright (c) 2021 Rop Gonggrijp. Based on esp_i2c_helper by Mika Tuupola.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+*/
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include <esp_log.h>
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/semphr.h"
+#include "freertos/task.h"
+#include <driver/i2c.h>
+
+#include "sdkconfig.h"
+
+#include "i2c_manager.h"
+
+
+#if defined __has_include
+	#if __has_include ("esp_idf_version.h")
+		#include "esp_idf_version.h"
+		#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4, 3, 0)
+			#define HAS_CLK_FLAGS
+		#endif
+	#endif
+#endif
```

Comment:
yes, the clk_flags in IDF 4.3 added some problems, but I think this solution is a little over-engineered.

This is what I was using and it compiles fine with or without `clk_flags` field in the config:

```
const i2c_config_t conf_master = {
        .mode = I2C_MODE_MASTER,
        .sda_pullup_en = GPIO_PULLUP_ENABLE,
        .scl_pullup_en = GPIO_PULLUP_ENABLE,
        .master.clk_speed = I2C_MASTER_FREQ_HZ,
        .sda_io_num = I2C_MASTER_SDA_IO,
        .scl_io_num = I2C_MASTER_SCL_IO
    };
```

The compiler will set all unset fields (clk_flags in IDF 4.3) to zero.

Reply:
If memory serves, it compiled without but then didn't work, giving me some obscure problem that I googled to be because of this.

Reply:
alright, lets stick on the safe side